container-chains/templates/frontier/node/src/service.rs

```diff
@@ -35,42 +38,78 @@ use {
 
 // Substrate Imports
 use {
+    fc_db::DatabaseSource,
+    fc_rpc_core::types::{FeeHistoryCache, FilterPool},
     sc_consensus::ImportQueue,
     sc_executor::NativeElseWasmExecutor,
     sc_network::NetworkBlock,
-    sc_service::{Configuration, PartialComponents, TFullBackend, TFullClient, TaskManager},
+    sc_service::{
+        BasePath, Configuration, PartialComponents, TFullBackend, TFullClient, TaskManager,
+    },
     sc_telemetry::{Telemetry, TelemetryWorker, TelemetryWorkerHandle},
 };
 
 /// Native executor type.
-pub struct ParachainNativeExecutor;
-
-impl sc_executor::NativeExecutionDispatch for ParachainNativeExecutor {
-    type ExtendHostFunctions = frame_benchmarking::benchmarking::HostFunctions;
-
-    fn dispatch(method: &str, data: &[u8]) -> Option<Vec<u8>> {
-        container_chain_template_frontier_runtime::api::dispatch(method, data)
-    }
-
-    fn native_version() -> sc_executor::NativeVersion {
-        container_chain_template_frontier_runtime::native_version()
-    }
-}
+use crate::client::TemplateRuntimeExecutor;
 
-type ParachainExecutor = NativeElseWasmExecutor<ParachainNativeExecutor>;
+pub type ParachainExecutor = NativeElseWasmExecutor<TemplateRuntimeExecutor>;
 
 type ParachainClient = TFullClient<Block, RuntimeApi, ParachainExecutor>;
 
 type ParachainBackend = TFullBackend<Block>;
 
-type ParachainBlockImport = TParachainBlockImport<Block, Arc<ParachainClient>, ParachainBackend>;
+use fc_consensus::FrontierBlockImport;
+
+pub fn frontier_database_dir(config: &Configuration, path: &str) -> std::path::PathBuf {
+    let config_dir = config
+        .base_path
+        .as_ref()
+        .map(|base_path| base_path.config_dir(config.chain_spec.id()))
+        .unwrap_or_else(|| {
+            BasePath::from_project("", "", "moonbeam").config_dir(config.chain_spec.id())
+        });
+    config_dir.join("frontier").join(path)
+}
+
+// TODO This is copied from frontier. It should be imported instead after
+// https://github.com/paritytech/frontier/issues/333 is solved
```

Comment:
That issue is still open but now that function is publicly available as `fc_db::Backend::open()`, so you can do something like

```rust
pub fn open_frontier_backend<C>(
    client: Arc<C>,
    config: &Configuration,
) -> Result<Arc<fc_db::Backend<Block>>, String>
where
    C: sp_blockchain::HeaderBackend<Block>,
{
    let config_dir = config
        .base_path
        .as_ref()
        .map(|base_path| base_path.config_dir(config.chain_spec.id()))
        .unwrap_or_else(|| {
            BasePath::from_project("", "", "moonbeam").config_dir(config.chain_spec.id())
        });
    let backend = fc_db::Backend::open(client.clone(), &config.database, &config_dir)?;

    Ok(Arc::new(backend))
}
```

Reply:
I did a similar thing to moonbeam where we have a different path depending on whether we have a rocksDb or a Parity db database. I think we should stay with that...